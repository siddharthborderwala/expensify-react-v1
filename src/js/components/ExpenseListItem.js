import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h4>{description}</h4>
		</Link>
		<p>
			&cent; {amount} - {createdAt}
		</p>
	</div>
);

export default ExpenseListItem;
