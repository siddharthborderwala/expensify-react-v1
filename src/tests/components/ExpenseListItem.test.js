import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from './../../js/components/ExpenseListItem';
import expenses, { expenseGenerator } from './../fixtures/expenses';

test('should render expense item with expense data', () => {
	const wrapper = shallow(
		<ExpenseListItem {...expenseGenerator('1', 'Coffee', '', 350, 0)} />
	);
	expect(wrapper).toMatchSnapshot();
});
