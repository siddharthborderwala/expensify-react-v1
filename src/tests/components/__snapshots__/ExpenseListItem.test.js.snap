// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item with expense data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h4>
      Coffee
    </h4>
  </Link>
  <p>
    Â¢ 
    350
     - 
    0
  </p>
</div>
`;
